pkgs/applications/virtualization/conmon/default.nix

```diff
@@ -0,0 +1,39 @@
+{ flavor ? ""
+, ldflags ? ""
+, stdenv
+, fetchFromGitHub
+, git
+, pkgconfig
+, glib
+, glibc
+, systemd
+}:
+
+stdenv.mkDerivation rec {
+  project = "conmon";
+  name = "${project}-${version}${flavor}";
+  version = "0.2.0";
+
+  src = fetchFromGitHub {
+    owner = "containers";
+    repo = project;
+    rev = "v${version}";
+    sha256 = "08fgkbv7hq62lcq39za9pm2s2j92ismgrkvfm7acwbvajqh9syjb";
+  };
+
+  nativeBuildInputs = [ git pkgconfig ];
+  buildInputs = [ glib systemd ] ++
+                stdenv.lib.optionals (glibc != null) [ glibc glibc.static ];
```

Comment:
What's the intent of checking if `glibc` is null? Why do we now need the `glibc.static` input? I assume you 1) want to make the binary static (which I suppose is fine, but should be handled with a flag IMO anyway or simply enabled unilaterally in all cases with a justification), or 2) make this buildable with Musl (but that won't be possible with this change anyway AFAICS, because the `systemd` buildInput is incompatible with musl as the libc. You really should instead be using `pkgsCross` or `pkgsMusl` to use `musl` as the libc on your end as a nixpkgs *user*, and this expression should instead check `stdenv.hostPlatform.isMusl` to conditionally include things in the `buildInputs` or not)

Reply:
The main reason behind this is to build it with musl libc and I managed to get systemd (or just the library part) to be built with musl. I changed the optional to `stdenv.hostPlatform.isMusl`.